Userland/Applications/FileManager/main.cpp

```diff
@@ -105,12 +106,39 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     // 3. the user's home directory
     // 4. the root directory
 
+    LexicalPath path(initial_location);
     if (!initial_location.is_empty()) {
         if (!ignore_path_resolution)
             initial_location = Core::DeprecatedFile::real_path_for(initial_location);
 
-        if (!Core::DeprecatedFile::is_directory(initial_location))
+        if (!Core::DeprecatedFile::is_directory(initial_location)) {
+            // We want to extract zips to a temporary directory when FileManager is launched with a .zip file as its first argument
+            if (path.has_extension(".zip"sv)) {
+                auto temp_directory = Core::TempFile::create_temp_directory();
+                if (temp_directory.is_error()) {
+                    dbgln("Failed to create temporary directory during zip extraction: {}", temp_directory.error());
+
+                    GUI::MessageBox::show_error(app->active_window(), "Failed to create temporary directory!"sv);
+                    return -1;
+                }
+
+                auto temp_directory_path = temp_directory.value()->path();
+                auto result = Core::Process::spawn("/bin/unzip"sv, Array { "-d"sv, temp_directory_path, initial_location });
```

Comment:
oh, and this will need to use the same mechanism as file copying to not block the UI thread for too long. Again, for the future :)

Reply:
Yeah, I was hoping to use some sort of fake-filesystem for ZIP files at some point, so I feel like this is fine until I think about how to implement something like that :)